cumulusci/tasks/push/tests/test_push_tasks.py

```diff
@@ -0,0 +1,481 @@
+        "state": "Released",
+    }
+    assert task._parse_version("1.2,Beta 3") == {
+        "major": "1",
+        "minor": "2",
+        "patch": None,
+        "build": "Beta 3",
+        "state": "Beta",
+    }
+    assert task._parse_version("1.2,(Beta 3)") == {
+        "major": "1",
+        "minor": "2",
+        "patch": None,
+        "build": "(Beta 3",
+        "state": "Beta",
+    }
+    assert task._parse_version("1.2") == {
+        "major": "1",
+        "minor": "2",
+        "patch": None,
+        "build": None,
+        "state": "Released",
+    }
+
+
+def test_get_version():
+    task = create_task(BaseSalesforcePushTask, options={})
+    assert task._get_version(
+        MetadataPackage(
+            push_api=mock.MagicMock(), name=NAME, sf_id=SF_ID, namespace=NAMESPACE
+        ),
+        "1.2.3.4",
+    )
+    assert task._get_version(
+        MetadataPackage(
+            push_api=mock.MagicMock(), name=NAME, sf_id=SF_ID, namespace=NAMESPACE
+        ),
+        "1.2,Beta 3",
+    )
+
+
+def test_get_version_error():
+    package = mock.MagicMock()
+    package.get_package_version_objs.return_value = None
+    task = create_task(BaseSalesforcePushTask, options={})
+    with pytest.raises(PushApiObjectNotFound):
+        task._get_version(package, VERSION)
+
+
+def test_get_package():
+    task = create_task(BaseSalesforcePushTask, options={})
+    task.push = mock.MagicMock()
+    task.push.get_package_objs.return_value = [
+        {"push_api": "123", "sf_id": SF_ID, "name": NAME, "namespace": NAMESPACE}
+    ]
+    assert task._get_package(NAMESPACE) == {
+        "push_api": "123",
+        "sf_id": SF_ID,
+        "name": NAME,
+        "namespace": NAMESPACE,
+    }
+
+
+def test_get_package_error():
+    task = create_task(BaseSalesforcePushTask, options={})
+    task.push = mock.MagicMock()
+    task.push.get_package_objs.return_value = None
+    with pytest.raises(PushApiObjectNotFound):
+        task._get_package(NAMESPACE)
+
+
+def test_schedule_push_org_list_get_orgs():
+    with open(ORG_FILE, "w") as file:
+        file.write(f"{ORG_FILE_TEXT}")
+    task = create_task(
+        SchedulePushOrgList,
+        options={
+            "orgs": ORG_FILE,
+            "version": VERSION,
+            "namespace": NAMESPACE,
+            "start_time": datetime.datetime.now(),
+            "batch_size": 10,
+        },
+    )
+    assert task._get_orgs() == ["00DS0000003TJJ6MAO", "00DS0000003TJJ6MAL"]
+    os.remove(ORG_FILE)
+
+
+def test_schedule_push_org_list_init_options():
+    with open(ORG_FILE, "w") as file:
+        file.write(f"{ORG_FILE_TEXT}")
+    task = create_task(
+        SchedulePushOrgList,
+        options={
+            "orgs": ORG_FILE,
+            "version": VERSION,
+            "start_time": datetime.datetime.now(),
+        },
+    )
+    task._init_task()
+    task._init_options(
+        {"orgs": ORG_FILE, "version": VERSION, "start_time": datetime.datetime.now()}
+    )
```

Comment:
_init_options is called by the task's `__init__` method when it is constructed; you don't need to call it again.